### DISCUSSION TITLE
Building blocks, components, microservice categories, etc...

### DISCUSSION THREAD

**Discussion by manjirisulekavadi** [Sunbird-Ed/Community#265] [Documentation] [answered]
In documentation, I am looking for a clean hierarchy of building blocks (16), components in it, microservice categories (7), microservices (19) and the APIs.
Where can I get a ready reference - especially of building blocks and the respective set of microservices? Microservices specific to the building block and/or used across the building blocks?

**Comment by Krishnaj20** [comment]
@manjirisulekavadi - I'd recommend you to go thru the Sunbird building blocks: https://sunbird.org/building-blocks . This page has links to the individual BB microsites as well.

**Reply by Krishnaj20** [reply]
@manjirisulekavadi - let us know if your query is answered.

**Reply by manjirisulekavadi** [reply]
@Krishnaj20 Thanks. Going through documentation from each microsites for every block. (block specific MS and APIs, those needed to interplay with other blocks, commonly used. - I think the Block definition is followed by the implementation of services and hence a bit confusing to a new / non-tech person)

Primarily I am looking for a ready reference or trying to create one for a documentation case like 
Building Block - SB Learn
Microservice Category - Registry
Microservices - User and Organization
APIs
-----
Create Organization
Read Organization Details
Search Org
Update Organization
Organization Update Status
Assign Key to an Organization

Create User V1, V2, V3, V4
Create SSOUser V1
User Sign up V1, V2
Create Managed User V1
Search User V1, V2, V3
Update User V1, V2, V3
Read User V1. V2, V3, V4, V5
Get User by ID Type
Check USer exists by Id type V2
Check USer exists by Id type V2
Block User
Unbloack User
Get User by Login Id
Get Managed User Details
Update User Declarations
User Migrate
User Merge
Assign Role V1, V2
Accept User TnC

**Reply by Krishnaj20** [reply]
@manjirisulekavadi - I'm not sure I understand your query. The microsites cover:
a) high level overview & functional capabilities of each building block
b) tech architecture
c) detailed product & developer guide of each component which is part of the building block
d) further details on installation, release cycles and release notes

**Comment by manjirisulekavadi** [comment, marked as answer]
Let me get back to you - going through all content
Thanks,
Manjiri Sule |  Asst. Vice President (Projects) | Tekdi Technologies Pvt.
Ltd.  |
Cell (+91) 9850961758 |
 Stay Green - Think Twice about the Need to Print and Avoid Printing

**Reply by Krishnaj20** [reply]
@manjirisulekavadi - We havent received any further information on this thread. Marking this as answered for now. Please feel free to raise a new one once you have all the details.
